torch/_dynamo/__init__.py

```diff
@@ -156,14 +164,126 @@ def mark_dynamic(t, index):
     4) Attempts to trace this function will explicitly raise. As such, all calls to mark_dynamic must be made
     before torch.compile.
 
+    """
+    mark_dynamic_constrain(t, index, min=None, max=None)
+
+
+@forbid_in_graph
+def mark_dynamic_constrain(
```

Comment:
I really don't like the lock-in to ranges on the one hand (e.g., using names like `dynamic_ranges`) and yet keeping this name generic (instead of, say, `mark_dynamic_ranges`). I do believe we need to add at least equality constraints in the short term, otherwise it doesn't feel complete. But if the argument is that "we'll get there when we get there" and potentially do further refactoring, then why not make it specific here too?

Just trying to understand the logic, I won't die on this hill. :)